ajax/timer.php

```diff
@@ -12,79 +12,128 @@
       case 'start':
          if (PluginActualtimeTask::checkTimerActive($task_id)) {
             $result=[
-               'mensage'=>__("A user is already performing the task", 'actualtime'),
-               'title' => __s('Warning'),
-               'class' => 'warn_msg',
+               'mensage' => __("A user is already performing the task", 'actualtime'),
+               'title'   => __('Warning'),
+               'class'   => 'warn_msg',
             ];
          } else {
-            if (PluginActualtimeTask::checkUserFree(Session::getLoginUserID())) {
+            if (! PluginActualtimeTask::checkUserFree(Session::getLoginUserID())) {
                $opcional=PluginActualtimeTask::getTicket(Session::getLoginUserID());
                $result=[
-                  'mensage'=>__("You are already doing a task", 'actualtime')." <a href='/front/ticket.form.php?id=".$opcional."'>".__("Ticket")."</a>",
-                  'title' => __s('Warning'),
-                  'class' => 'warn_msg',
+                  'mensage' => __("You are already doing a task", 'actualtime')." <a href='/front/ticket.form.php?id=".$opcional."'>".__("Ticket")."&forcetab=Ticket%241</a>",
+                  'title'   => __('Warning'),
+                  'class'   => 'warn_msg',
                ];
             } else {
-               $DB->insert(
-                  'glpi_plugin_actualtime_tasks', [
-                     'tasks_id'      => $task_id,
-                     'actual_begin' => date("Y-m-d H:i:s"),
-                     'users_id'=>Session::getLoginUserID(),
-                  ]
-               );
-               $result=[
-                  'mensage'=>__("Timer started", 'actualtime'),
-                  'title' => __s('Information'),
-                  'class' => 'info_msg',
-               ];
+               if (! PluginActualtimeTask::checkTech($task_id)) {
```

Comment:
This check is redundant.
It is already checked when you show the buttons
Please remove it

Reply:
Yes, it is redundant, as stated in the code itself. I was just cautious if someone could use the ajax resquest to do something wrong, so double checked any database modification. Anyway, you are right. It is just checking twice. Branch was corrected with commit [4e60eb8](https://github.com/Multiconecta/actualtime/commit/4e60eb8dc1288a732df7bb50d8c91a6a0ea236e5)